src/Orchard.Web/Modules/Orchard.Workflows/Views/Admin/Edit.cshtml

```diff
@@ -74,23 +75,41 @@
 @Html.Hidden("data", String.Empty)
 
 @Html.Hidden("confirm-delete-activity", T("Are you sure you want to remove this activity?"))
-@Html.Hidden("confirm-delete-instances", T("Are you sure you want to remove running instances of this workflow?"))
+@Html.Hidden("confirm-delete-instances", T("You have running instances of this workflow, do you want to stop them?"))
 
     using (Script.Foot()) {
 <script type="text/javascript">
     //<![CDATA[
-    $("form").submit(function () {
-        saveLocal(localId);
-        var workflow = loadWorkflow(localId);
-        var data = JSON.stringify(workflow);
-        $("[name='data']").val(data);
+    $("form").submit(function (e, submit, clearWorkflows) {
+        if(submit){
+            saveLocal(localId);
+            var workflow = loadWorkflow(localId);
+            var data = JSON.stringify(workflow);
+            $("[name='data']").val(data);
+            var values = [$("<input>", { type: "hidden", name: "clearWorkflows", value: clearWorkflows }), $("<input>", { type: "hidden", name: "submit.Save", value: "Save" })];
+            $(this).append(values);
+            return true;
+        }
+
+        e.preventDefault();
 
         $.ajax({
             url: stateUrl + "/" + $("#id").val(),
             async: false,
             success: function(state) {
-                if(state.isRunning && !confirm($("#confirm-delete-instances").val())) {
-                    e.preventDefault();
+                if (state.isRunning) {
+                    var dialog = $('<p>' + $("#confirm-delete-instances").val() + '</p>').dialog({
+                        buttons: {
+                            '@T("Yes")': function() { $('form').trigger('submit', [true, true]); },

```

Comment:
This also needs to be JavascriptEncoded, but that's fine for now. Can be done in another PR.